test/anvil/entrypointsimulations.test.ts

```diff
@@ -0,0 +1,252 @@
+      const op = await fillAndSign({
+        preVerificationGas: BigNumber.from(2).pow(130),
+        sender: account1.address
+      }, accountOwner1, entryPoint)
+      await expect(
+        simulateValidation(op, entryPoint.address)
+      ).to.revertedWith('gas values overflow')
+    })
+
+    it('should fail creation for wrong sender', async () => {
+      const op1 = await fillAndSign({
+        initCode: getAccountInitCode(accountOwner1.address, simpleAccountFactory),
+        sender: '0x'.padEnd(42, '1'),
+        verificationGasLimit: 30e6
+      }, accountOwner1, entryPoint)
+      await expect(simulateValidation(op1, entryPoint.address))
+        .to.revertedWith('AA14 initCode must return sender')
+    })
+
+    it('should report failure on insufficient verificationGas (OOG) for creation', async () => {
+      const initCode = getAccountInitCode(accountOwner1.address, simpleAccountFactory)
+      const sender = await entryPoint.callStatic.getSenderAddress(initCode).catch(e => e.errorArgs.sender)
+      const op0 = await fillAndSign({
+        initCode,
+        sender,
+        verificationGasLimit: 5e5,
+        maxFeePerGas: 0
+      }, accountOwner1, entryPoint)
+      // must succeed with enough verification gas.
+      await simulateValidation(op0, entryPoint.address, { gas: '0xF4240' })
+
+      const op1 = await fillAndSign({
+        initCode,
+        sender,
+        verificationGasLimit: 1e5,
+        maxFeePerGas: 0
+      }, accountOwner1, entryPoint)
+      await expect(simulateValidation(op1, entryPoint.address, { gas: '0xF4240' }))
+        .to.revertedWith('AA13 initCode failed or OOG')
+    })
+
+    it('should succeed for creating an account', async () => {
+      const sender = await getAccountAddress(accountOwner1.address, simpleAccountFactory)
+      const op1 = await fillAndSign({
+        sender,
+        initCode: getAccountInitCode(accountOwner1.address, simpleAccountFactory)
+      }, accountOwner1, entryPoint)
+      await fund(op1.sender)
+
+      await simulateValidation(op1, entryPoint.address)
+    })
+
+    it('should not call initCode from entrypoint', async () => {
+      // a possible attack: call an account's execFromEntryPoint through initCode. This might lead to stolen funds.
+      const { proxy: account } = await createAccount(ethersSigner, await accountOwner.getAddress(), entryPoint.address)
+      const sender = createAddress()
+      const op1 = await fillAndSign({
+        initCode: hexConcat([
+          account.address,
+          account.interface.encodeFunctionData('execute', [sender, 0, '0x'])
+        ]),
+        sender
+      }, accountOwner, entryPoint)
+      const error = await simulateValidation(op1, entryPoint.address).catch(e => e)
+      expect(error.message).to.match(/initCode failed or OOG/, error)
+    })
+
+    // TODO: this test is impossible to do with the "state override" approach
```

Comment:
since anvil does support debug_traceCall, it is possible to write it better.
but this test is very partial.
should instead say:
for actual opcode and storage rule restrictions, see the reference bundler ValidationManager

Reply:
Added comment.